### PR TITLE
Post creator

### PR DESCRIPTION
**Pull request by DavidKalajdzic**
From the page of an event we can now create a post associated to this particular event. A post in mainly composed of a title, description, set of images, visibility setting (public vs private)

If the creation of a post fails an error message is displayed to the user.

### CODE REVIEW

File: app/src/androidTest/java/com/sdpteam/connectout/remoteStorage/FileStorageFirebaseTest.java [hunk 1]
```diff
@@ -38,4 +40,21 @@ public void testUploadImageToFirebase() {
...
+        Resources resources = intentsTestRule.getActivity().getResources();
+
+        // Create the resource Uri for the event_image drawable
+        Uri uri = Uri.parse(ContentResolver.SCHEME_ANDROID_RESOURCE + "://" +
+                resources.getResourcePackageName(R.drawable.event_image) + '/' +
+                resources.getResourceTypeName(R.drawable.event_image) + '/' +
+                resources.getResourceEntryName(R.drawable.event_image));
+        if (uri == null) {
+            throw new IllegalStateException("Why is it null?");
+        }
```
Comment (MikaelKalajdzic) [inline app/src/androidTest/java/com/sdpteam/connectout/remoteStorage/FileStorageFirebaseTest.java:55]: I would go for assertNotNull(uri) simply, but ok

File: app/src/main/java/com/sdpteam/connectout/event/EventFirebaseDataSource.java [hunk 2]
```diff
@@ -110,10 +110,12 @@ public CompletableFuture<Boolean> leaveEvent(String eventId, String participantI
      */
     @Override
     public CompletableFuture<Event> getEvent(String eventId) {
+        System.out.println("Fetching event from firebase: " + eventId);
```
Comment (MikaelKalajdzic) [inline app/src/main/java/com/sdpteam/connectout/event/EventFirebaseDataSource.java:113]: I don't know if it the good way to log. Ok for me

File: app/src/main/java/com/sdpteam/connectout/post/model/PostFirebaseDataSource.java [hunk 3]
```diff
@@ -117,8 +117,11 @@ public CompletableFuture<Result<List<Post>>> fetchAllPosts(String userId) {
                     final List<Post> postList = StreamSupport.stream(dataSnapshot.getChildren().spliterator(), false)
                             .map(childSnapshot -> childSnapshot.getValue(Post.class))
                             .filter(Objects::nonNull)
+                            .filter(post -> post.getProfileId() != null)
+                            .filter(post -> post.getEventId() != null)
+                            .filter(post -> post.getVisibility() != null)
                             .filter(post -> {
-                                if (post.isSemiPrivate()) {
+                                if (post.getVisibility().equals(SEMIPRIVATE)) {
```
Comment (MikaelKalajdzic) [inline app/src/main/java/com/sdpteam/connectout/post/model/PostFirebaseDataSource.java:124]: Why is that cute isSemiPrivate method removed?

File: app/src/main/java/com/sdpteam/connectout/post/view/PostCreatorActivity.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+import android.widget.Switch;
+import android.widget.TextView;
+import androidx.annotation.IdRes;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.appcompat.widget.Toolbar;
+import androidx.fragment.app.FragmentTransaction;
+
+public class PostCreatorActivity extends AppCompatActivity {
+
+    PostCreatorViewModel viewModel;
```
Comment (MikaelKalajdzic) [inline app/src/main/java/com/sdpteam/connectout/post/view/PostCreatorActivity.java:25]: Shouldn't it be private ?

File: app/src/main/java/com/sdpteam/connectout/post/view/PostCreatorActivity.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+     * @param fromContext from where we are starting the intent
+     */
+    public static void openPostCreator(Context fromContext, String profileId, String eventId, String eventName) {
+        Intent intent = new Intent(fromContext, PostCreatorActivity.class);
+        intent.putExtra("profileId", profileId);
+        intent.putExtra("eventId", eventId);
+        intent.putExtra("eventName", eventName);
+        fromContext.startActivity(intent);
+    }
+}
```
Comment (MikaelKalajdzic) [inline app/src/main/java/com/sdpteam/connectout/post/view/PostCreatorActivity.java:107]: Nice activity

File: app/src/main/java/com/sdpteam/connectout/utils/MyListView.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
+package com.sdpteam.connectout.utils;
+
+import android.content.Context;
+import android.util.AttributeSet;
+import android.view.View;
+import android.widget.ListView;
+
+// Copied from https://stackoverflow.com/questions/11295080/android-wrap-content-is-not-working-with-listview
+public class MyListView extends ListView {
```
Comment (MikaelKalajdzic) [inline app/src/main/java/com/sdpteam/connectout/utils/MyListView.java:9]: I would give it a better name, but I agree is not easy to find...

**Review by MikaelKalajdzic — APPROVED** [review]
Niiiiiiiice
I like the UI
I approve, but still if you have time to correct 2-3 little details it'll be cool

**Review by MikaelKalajdzic — APPROVED** [review]

File: app/src/main/java/com/sdpteam/connectout/post/view/PostCreatorActivity.java (hunk 4 above)
Comment (DavidKalajdzic) [reply app/src/main/java/com/sdpteam/connectout/post/view/PostCreatorActivity.java:25]: It should but unfortunately writing tests would be much harder

File: app/src/androidTest/java/com/sdpteam/connectout/remoteStorage/FileStorageFirebaseTest.java (hunk 1 above)
Comment (DavidKalajdzic) [reply app/src/androidTest/java/com/sdpteam/connectout/remoteStorage/FileStorageFirebaseTest.java:55]: It was just for the test, no big deal

**Pull request merged**